torch/csrc/autograd/autograd_not_implemented_fallback.cpp

```diff
@@ -0,0 +1,180 @@
+  const auto& schema = op.schema();
+  const auto& op_name = schema.operator_name().name;
+  const auto& arguments = schema.arguments();
+  const auto& returns = schema.returns();
+  const auto num_arguments = arguments.size();
+  const auto num_returns = returns.size();
+  const auto stack_start = stack->size() - num_arguments;
+  const bool grad_mode = GradMode::is_enabled();
+  std::vector<const at::Tensor*> tensors_requiring_grad_on_stack;
+
+  // Keep track of which outputs are output of in-place modification
+  // so we can rebase_history if necessary
+  std::vector<bool> is_inplace_output;
+  std::vector<bool> is_aliased_output;
+  is_inplace_output.reserve(num_returns);
+  is_aliased_output.reserve(num_returns);
+
+  for (const auto i : c10::irange(num_returns)) {
+    const auto& alias_info = returns[i].alias_info();
+    is_inplace_output.push_back(alias_info.has_value() && alias_info->isWrite());
+    is_aliased_output.push_back(alias_info.has_value());
+
+  }
+
+  #ifndef NDEBUG
+  int aliased_input_idx = -1;
+  int aliased_output_idx = -1;
+  for (const auto i : c10::irange(num_returns)) {
+    const auto& alias_info = returns[i].alias_info();
+    if (alias_info.has_value()) {
+      AT_ASSERT(aliased_output_idx == -1); // Assume only a single aliased output
+      aliased_output_idx = i;
+    }
+  }
+  for (const auto i : c10::irange(num_arguments)) {
+    const auto& alias_info = arguments[i].alias_info();
+    if (alias_info.has_value()) {
+      AT_ASSERT(aliased_input_idx == -1); // Assume only a single aliased input
+      aliased_input_idx = i;
+    }
+  }
+  #endif
+
+  size_t num_tensor_inputs = 0;  // Only used for DEBUG-only checks
+
+  _foreach_tensor([&](size_t _, size_t idx_arg, const at::Tensor& t) {
+    if (arguments[idx_arg].type()->kind() != c10::TypeKind::OptionalType) {
+      TORCH_CHECK(t.defined(), "Expected argument ", idx_arg, " of ", op_name, " to be defined.");
+    }
+    if (grad_mode && t.requires_grad()) {
+      tensors_requiring_grad_on_stack.push_back(&t);
+    }
+    num_tensor_inputs++;
+    TORCH_CHECK_NOT_IMPLEMENTED(!isFwGradDefined(t), "Trying to use forward AD with ", op_name, " that does not support it.");
+  }, stack, stack_start, num_arguments);
+
+  const bool any_requires_grad = tensors_requiring_grad_on_stack.size() > 0;
+
+  _foreach_tensor([&](size_t _, size_t i, const at::Tensor& t) {
+    const auto& alias_info = arguments[i].alias_info();
+    if (alias_info.has_value() && alias_info->isWrite()) {
+      check_inplace(t, any_requires_grad);
+    }
+  }, stack, stack_start, num_arguments);
+
+  std::shared_ptr<NotImplemented> grad_fn;
+  if (any_requires_grad) {
+    grad_fn = std::shared_ptr<NotImplemented>(new NotImplemented(op_name), deleteNode);
+    grad_fn->set_next_edges(collect_next_edges(tensors_requiring_grad_on_stack));
+  }
+
+  #ifndef NDEBUG
+  // See NOTE [ TensorImpl and Storage Pointer Sanity Checks ]
+  std::vector<c10::IValue> stack_copy(*stack);
```

Comment:
It's a bad idea to copy the entire stack, as you may be arbitrarily deep in the stack, and this would copy all of the parent frames as well. Just copy the argument segment of the stack.